crates/adapters/src/integrated/delta_table/test.rs

```diff
@@ -141,6 +145,35 @@ async fn wait_for_count_records(
     }
 }
 
+/// Wait until materialized table contains exactly `expected_count` records.
+async fn wait_for_count_records_materialized(
```

Comment:
very strange function, including the hardwired timeout

Reply:
It's part of the test harness, with a generous timeout to prevent the failing test from getting stuck in CI.